File: demo/src/main/java/com/example/demo/models/Measurement.java
package com.example.demo.models;

import lombok.*;

import javax.persistence.*;

@Data
@Entity(name = "measurement")
public class Measurement {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "Id")
    private int id;
    @ManyToOne
    @JoinColumn(name = "ExaminationDateId", referencedColumnName = "id")
    private ExaminationDate examinationDateId;
    @ManyToOne
    @JoinColumn(name = "OrganizationId", referencedColumnName = "id")
    private Organization organizationId;
    @Basic
    @Column(name = "LineNo")
    private Integer lineNo;
    @Basic
    @Column(name = "LandCode")
    private Integer landCode;
    @Basic
    @Column(name = "ProductionAreaNo")
    private Integer productionAreaNo;
    @Basic
    @Column(name = "CropRotationNo")
    private Integer cropRotationNo;
    @Basic
    @Column(name = "CropRotationFieldNo")
    private Integer cropRotationFieldNo;
    @Basic
    @Column(name = "CropRotationFieldArea")
    private Integer cropRotationFieldArea;
    @Basic
    @Column(name = "WorkingAreaNo")
    private Integer workingAreaNo;
    @Basic
    @Column(name = "WorkingAreaArea")
    private Integer workingAreaArea;
    @Basic
    @Column(name = "ElementaryAreaNo")
    private Integer elementaryAreaNo;
    @Basic
    @Column(name = "ElementaryAreaArea")
    private Integer elementaryAreaArea;
    @ManyToOne
    @JoinColumn(name = "AmeliorativeStateId", referencedColumnName = "id")
    private AmeliorativeState ameliorativeStateId;
    @Basic
    @Column(name = "HumusHorizonThickness")
    private Integer humusHorizonThickness;
    @Basic
    @Column(name = "pH")
    private Integer pH;
    @Basic
    @Column(name = "Humus")
    private Integer humus;
    @Basic
    @Column(name = "Measurement_P2O5")
    private Integer measurementP2O5;
    @Basic
    @Column(name = "Measurement_K2O")
    private Integer measurementK2O;
    @Basic
    @Column(name = "Measurement_CaO")
    private Integer measurementCaO;
    @Basic
    @Column(name = "Measurement_MgO")
    private Integer measurementMgO;
    @Basic
    @Column(name = "Measurement_Sulphur")
    private Integer measurementSulphur;
    @Basic
    @Column(name = "Measurement_Boron")
    private Integer measurementBoron;
    @Basic
    @Column(name = "Measurement_Copper")
    private Integer measurementCopper;
    @Basic
    @Column(name = "Measurement_Zink")
    private Integer measurementZink;
    @Basic
    @Column(name = "Measurement_Manganese")
    private Integer measurementManganese;
    @Basic
    @Column(name = "Measurement_Cobalt")
    private Integer measurementCobalt;
    @Basic
    @Column(name = "CationExchangeCapacity")
    private Integer cationExchangeCapacity;
    @Basic
    @Column(name = "CationExchangeCapacityK20")
    private Integer cationExchangeCapacityK20;
    @Basic
    @Column(name = "P2O5InExtract")
    private Integer p2O5InExtract;
    @Basic
    @Column(name = "Measurement_NO3_An")
    private Integer measurementNo3An;
    @Basic
    @Column(name = "Measurement_NO3_A2")
    private Integer measurementNo3A2;
    @Basic
    @Column(name = "Measurement_NO4Plus_An")
    private Integer measurementNo4PlusAn;
    @Basic
    @Column(name = "Measurement_NO4Plus_A2")
    private Integer measurementNo4PlusA2;
    @Basic
    @Column(name = "Measurement_Lead")
    private Integer measurementLead;
    @Basic
    @Column(name = "Measurement_Cadmium")
    private Integer measurementCadmium;
    @Basic
    @Column(name = "GrossContent_Lead")
    private Integer grossContentLead;
    @Basic
    @Column(name = "GrossContent_Cadmium")
    private Integer grossContentCadmium;
    @Basic
    @Column(name = "GrossContent_Copper")
    private Integer grossContentCopper;
    @Basic
    @Column(name = "GrossContent_Zink")
    private Integer grossContentZink;
    @Basic
    @Column(name = "GammaBackground")
    private Integer gammaBackground;
    @Basic
    @Column(name = "Radionuclides_Cesium")
    private Integer radionuclidesCesium;
    @Basic
    @Column(name = "Radionuclides_Strontium")
    private Integer radionuclidesStrontium;
    @ManyToOne
    @JoinColumn(name = "SoilTypeId", referencedColumnName = "id")
    private SoilType soilTypeId;
    @ManyToOne
    @JoinColumn(name = "SoilHydrationAndTurfId", referencedColumnName = "id")
    private SoilHydrationAndTurf soilHydrationAndTurfId;
    @ManyToOne
    @JoinColumn(name = "SoilGranulometricCompositionId", referencedColumnName = "id")
    private SoilGranulometricComposition soilGranulometricCompositionId;
    @ManyToOne
    @JoinColumn(name = "SoilBedrockId", referencedColumnName = "id")
    private SoilBedrock soilBedrockId;
    @ManyToOne
    @JoinColumn(name = "SoilParentrockId", referencedColumnName = "id")
    private SoilParentrock soilParentrockId;
}

File: demo/src/main/resources/static/js/entity_descriptions/measurement.js
export const entityDescription = [
    {
        name: "ID",
        getter: e => e["id"],
        prop: "id"
    },
    {
        name: "Тур",
        getter: e => e["examinationDateId"]["tour"],
        prop: "examinationDateId.tour"
    },
    {
        name: "Год",
        getter: e => e["examinationDateId"]["examinationYear"],
        prop: "examinationDateId.examinationYear"
    },
    {
        name: "Наименование организации",
        getter: e => e["organizationId"]["name"],
        prop: "organizationId.name"
    },
    {
        name: "Код угодья",
        getter: e => e["landCode"],
        prop: "landCode"
    },
    {
        name: "Hомеp пpоизводственного участка",
        getter: e => e["productionAreaNo"],
        prop: "productionAreaNo"
    },
    {
        name: "Hомеp севообоpота",
        getter: e => e["cropRotationNo"],
        prop: "cropRotationNo"
    },
    {
        name: "Hомеp поля севообоpота",
        getter: e => e["cropRotationFieldNo"],
        prop: "cropRotationFieldNo"
    },
    {
        name: "Площадь поля севообоpота",
        getter: e => e["cropRotationFieldArea"],
        prop: "cropRotationFieldArea"
    },
    {
        name: "Hомеp pабочего участка",
        getter: e => e["workingAreaNo"],
        prop: "workingAreaNo"
    },
    {
        name: "Площадь pабочего участка",
        getter: e => e["workingAreaArea"],
        prop: "workingAreaArea"
    },
    {
        name: "Hомеp элементаpного участка",
        getter: e => e["elementaryAreaNo"],
        prop: "elementaryAreaNo"
    },
    {
        name: "Площадь элементаpного участка",
        getter: e => e["elementaryAreaArea"],
        prop: "elementaryAreaArea"
    },
    {
        name: "Тип почвы",
        getter: e => e["soilTypeId"]["meaning"],
        prop: "soilTypeId.meaning"
    },
    {
        name: "Степень увлажнения, мощность торфяных залежей",
        getter: e => e["soilHydrationAndTurfId"]["meaning"],
        prop: "soilHydrationAndTurfId.meaning"
    },
    {
        name: "Гранулометрический состав",
        getter: e => e["soilGranulometricCompositionId"]["meaning"],
        prop: "soilGranulometricCompositionId.meaning"
    },
    {
        name: "Подстилающая порода",
        getter: e => e["soilBedrockId"]["meaning"],
        prop: "soilBedrockId.meaning"
    },
    {
        name: "Почвообразующая порода",
        getter: e => e["soilParentrockId"]["meaning"],
        prop: "soilParentrockId.meaning"
    },

    {
        name: "Мелиоpативное состояние",
        getter: e => e["ameliorativeStateId"]["meaning"],
        prop: "ameliorativeStateId.meaning"
    },
    {
        name: "Мощность гумусового гоpизонта",
        getter: e => e["humusHorizonThickness"],
        prop: "humusHorizonThickness"
    },
    {
        name: "pH в KCL",
        getter: e => e["ph"],
        prop: "ph"
    },
    {
        name: "Гумус",
        getter: e => e["humus"],
        prop: "humus"
    },
    {
        name: "СППФ P2O5, мг/кг",
        getter: e => e["measurementP2O5"],
        prop: "measurementP2O5"
    },
    {
        name: "СППФ K2O, мг/кг",
        getter: e => e["measurementK2O"],
        prop: "measurementK2O"
    },
    {
        name: "СППф CaO, мг/кг",
        getter: e => e["measurementCaO"],
        prop: "measurementCaO"
    },
    {
        name: "СППФ MgO, мг/кг",
        getter: e => e["measurementMgO"],
        prop: "measurementMgO"
    },
    {
        name: "СППФ серы, мг/кг",
        getter: e => e["measurementSulphur"],
        prop: "measurementSulphur"
    },
    {
        name: "СППФ бора, мг/кг",
        getter: e => e["measurementBoron"],
        prop: "measurementBoron"
    },
    {
        name: "СППФ меди, мг/кг",
        getter: e => e["measurementCopper"],
        prop: "measurementCopper"
    },
    {
        name: "СППФ цинка, мг/кг",
        getter: e => e["measurementZink"],
        prop: "measurementZink"
    },
    {
        name: "СППФ Mn, мг/кг",
        getter: e => e["measurementManganese"],
        prop: "measurementManganese"
    },
    {
        name: "СППФ Co, мг/кг",
        getter: e => e["measurementCobalt"],
        prop: "measurementCobalt"
    },
    {
        name: "Емкость катионного обмена",
        getter: e => e["cationExchangeCapacity"],
        prop: "cationExchangeCapacity"
    },
    {
        name: "% K2O от емкости катионного обмена",
        getter: e => e["cationExchangeCapacityK20"],
        prop: "cationExchangeCapacityK20"
    },
    {
        name: "P2O5 в вытяжке 0,01 MCaCl2, мг/л",
        getter: e => e["p2O5InExtract"],
        prop: "p2O5InExtract"
    },
    {
        name: "СППФ NO3 An, мг/кг",
        getter: e => e["measurementNo3An"],
        prop: "measurementNo3An"
    },
    {
        name: "СППФ NO3 A2, мг/кг",
        getter: e => e["measurementNo3A2"],
        prop: "measurementNo3A2"
    },
    {
        name: "СППФ NH4+ An, мг/кг",
        getter: e => e["measurementNo4PlusAn"],
        prop: "measurementNo4PlusAn"
    },
    {
        name: "СППФ NH4+ A2, мг/кг",
        getter: e => e["measurementNo4PlusA2"],
        prop: "measurementNo4PlusA2"
    },
    {
        name: "СППФ Pb, мг/кг",
        getter: e => e["measurementLead"],
        prop: "measurementLead"
    },
    {
        name: "СППФ Cd, мг/кг",
        getter: e => e["measurementCadmium"],
        prop: "measurementCadmium"
    },
    {
        name: "Валовое содеpжание Рв,мг/кг",
        getter: e => e["grossContentLead"],
        prop: "grossContentLead"
    },
    {
        name: "Валовое содеpжание Cd,мг/кг",
        getter: e => e["grossContentCadmium"],
        prop: "grossContentCadmium"
    },
    {
        name: "Валовое содеpжание Сu,мг/кг",
        getter: e => e["grossContentCopper"],
        prop: "grossContentCopper"
    },
    {
        name: "Валовое содеpжание Zn,мг/кг",
        getter: e => e["grossContentZink"],
        prop: "grossContentZink"
    },
    {
        name: "Гамма-фон, мкp/час",
        getter: e => e["gammaBackground"],
        prop: "gammaBackground"
    },
    {
        name: "Радионуклиды Cs, ки/кв.км",
        getter: e => e["radionuclidesCesium"],
        prop: "radionuclidesCesium"
    },
    {
        name: "Радионуклиды Sr, ки/кв.км",
        getter: e => e["radionuclidesStrontium"],
        prop: "radionuclidesStrontium"
    },
]